packages/contracts/src/connect-contracts.ts

```diff
@@ -0,0 +1,136 @@
+import { getL1ContractData, getL2ContractData } from './contract-data'
+
+export const connectL1Contracts = async (
+  signerOrProvider: Signer | providers.Provider,
+  network: 'goerli' | 'kovan' | 'mainnet'
+): Promise<Object> => {
+  const l1ContractData = getL1ContractData(network)
+
+  return {
+    addressManager: new Contract(
+      l1ContractData.Lib_AddressManager.abi,
+      l1ContractData.Lib_AddressManager.address,
+      signerOrProvider
+    ),
+    canonicalTransactionChain: new Contract(
+      l1ContractData.OVM_CanonicalTransactionChain.abi,
+      l1ContractData.OVM_CanonicalTransactionChain.address,
+      signerOrProvider
+    ),
+    executionManager: new Contract(
+      l1ContractData.OVM_ExecutionManager.abi,
+      l1ContractData.OVM_ExecutionManager.address,
+      signerOrProvider
+    ),
+    fraudVerifier: new Contract(
+      l1ContractData.OVM_FraudVerifier.abi,
+      l1ContractData.OVM_FraudVerifier.address,
+      signerOrProvider
+    ),
+    xDomainMessenger: new Contract(
+      l1ContractData.OVM_L1CrossDomainMessenger.abi,
+      l1ContractData.OVM_L1CrossDomainMessenger.address,
+      signerOrProvider
+    ),
+    ethGateway: new Contract(
+      l1ContractData.OVM_L1ETHGateway.abi,
+      l1ContractData.OVM_L1ETHGateway.address,
+      signerOrProvider
+    ),
+    multiMessageRelayer: new Contract(
+      l1ContractData.OVM_L1MultiMessageRelayer.abi,
+      l1ContractData.OVM_L1MultiMessageRelayer.address,
+      signerOrProvider
+    ),
+    safetyChecker: new Contract(
+      l1ContractData.OVM_SafetyChecker.abi,
+      l1ContractData.OVM_SafetyChecker.address,
+      signerOrProvider
+    ),
+    stateCommitmentChain: new Contract(
+      l1ContractData.OVM_StateCommitmentChain.abi,
+      l1ContractData.OVM_StateCommitmentChain.address,
+      signerOrProvider
+    ),
+    stateManagerFactory: new Contract(
+      l1ContractData.OVM_StateManagerFactory.abi,
+      l1ContractData.OVM_StateManagerFactory.address,
+      signerOrProvider
+    ),
+    stateTransitionerFactory: new Contract(
+      l1ContractData.OVM_StateTransitionerFactory.abi,
+      l1ContractData.OVM_StateTransitionerFactory.address,
+      signerOrProvider
+    ),
+    xDomainMessengerProxy: new Contract(
+      l1ContractData.Proxy__OVM_L1CrossDomainMessenger.abi,
+      l1ContractData.Proxy__OVM_L1CrossDomainMessenger.address,
+      signerOrProvider
+    ),
+    l1EthGatewayProxy: new Contract(
+      l1ContractData.Proxy__OVM_L1ETHGateway.abi,
+      l1ContractData.Proxy__OVM_L1ETHGateway.address,
+      signerOrProvider
+    ),
+    mockBondManger: new Contract(
+      l1ContractData.mockOVM_BondManager.abi,
+      l1ContractData.mockOVM_BondManager.address,
+      signerOrProvider
+    ),
+  }
+}
+
+export const connectL2Contracts = async (
+  signerOrProvider: Signer | providers.Provider
+): Promise<Object> => {
+  const l2ContractData = getL2ContractData()
+
+  return {
+    eth: new Contract(
```

Comment:
I'm unsure about naming. I think it's probably good to shorten for convenience and remove the "L1" and "L2" from contract names since the user will be connecting to each layer with separate functions?  

But it might add confusion to stray from the .sol names. ex: `OVM_L2CrossDomainMessenger` -> `xDomainMessenger`

Reply:
Good questions... would like @smartcontracts's input.